PRs from forks fail Travis CI with `Earthdata authentication failed`

In #271 we ran into the issue of all Travis-CI tests erroring after the setup in `conftest.py` failed because of secret environment variables. Travis-CI does not allow use of encrypted env variables for external (fork-based) PRs because it's a security risk. We need to modify our CI setup ([e.g.](https://docs.travis-ci.com/user/pull-requests/#pull-requests-and-security-restrictions)) so that tests that don't require the env variable can still run for forked PRs.

Example Travis CI failure from a fork PR:

```
FAILED icepyx/tests/test_auth.py::test_get_session - icepyx.core.auth.AuthenticationError: Earthdata authentication failed. Check output for error message

FAILED icepyx/tests/test_auth.py::test_get_s3login_credentials - icepyx.core.auth.AuthenticationError: Earthdata authentication failed. Check output for error message

FAILED icepyx/tests/test_auth.py::test_login_function - icepyx.core.auth.AuthenticationError: Earthdata authentication failed. Check output for error message
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -27,7 +27,7 @@ stages:
 jobs:
   include:
     - stage: basic tests
-      script: pytest icepyx/ --verbose --cov app --ignore icepyx/tests/test_behind_NSIDC_API_login.py
+      script: pytest icepyx/ --verbose --cov app --ignore=icepyx/tests/test_behind_NSIDC_API_login.py --ignore=icepyx/tests/test_auth.py
       # includes an integrity check of the uploader as recommended by CodeCov
       after_success:
         - curl https://keybase.io/codecovsecurity/pgp_keys.asc | gpg --no-default-keyring --keyring trustedkeys.gpg --import # One-time step
@@ -41,4 +41,4 @@ jobs:
     - stage: behind Earthdata
       script:
         - export EARTHDATA_PASSWORD=$NSIDC_LOGIN
-        - pytest icepyx/tests/test_behind_NSIDC_API_login.py
+        - pytest icepyx/tests/test_behind_NSIDC_API_login.py icepyx/tests/test_auth.py

